roles/foreman/tasks/main.yaml

```diff
@@ -0,0 +1,114 @@
+---
+- name: Pull the Foreman container image
+  containers.podman.podman_image:
+    name: "{{ foreman_image }}"
+    state: present
+
+- name: Create secret for DATABASE_URL
+  containers.podman.podman_secret:
+    state: present
+    name: foreman-database-url
+    data: "postgresql://foreman:{{ foreman_db_password }}@localhost/foreman"
+
+- name: Create settings config secret
+  containers.podman.podman_secret:
+    state: present
+    name: foreman-settings-yaml
+    data: "{{ lookup('ansible.builtin.template', 'settings.yaml.j2') }}"
+
+- name: Create katello config secret
+  containers.podman.podman_secret:
+    state: present
+    name: foreman-katello-yaml
+    data: "{{ lookup('ansible.builtin.template', 'katello.yaml.j2') }}"
+
+- name: Create the podman secret for Candlepin client certificate
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-client-cert
+    path: "{{ candlepin_client_certificate }}"
+
+- name: Create the podman secret for Candlepin client key
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-client-key
+    path: "{{ candlepin_client_key }}"
+
+- name: Create the podman secret for Candlepin tomcat certificate
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-tomcat-cert
+    path: "{{ candlepin_tomcat_certificate }}"
+
+- name: Create the podman secret for Candlepin tomcat key
+  containers.podman.podman_secret:
+    state: present
+    name: candlepin-tomcat-key
+    path: "{{ candlepin_tomcat_key }}"
+
+- name: Deploy Foreman Container
+  containers.podman.podman_container:
+    name: "foreman"
+    image: "{{ foreman_image }}"
+    state: quadlet
+    ports: "{{ foreman_ports }}"
+    network: host
+    hostname: "{{ ansible_fqdn }}"
+    secrets:
+      - 'foreman-database-url,type=env,target=DATABASE_URL'
+      - 'foreman-settings-yaml,type=mount,target=/etc/foreman/settings.yaml'
+      - 'foreman-katello-yaml,type=mount,target=/etc/foreman/plugins/katello.yaml'
+      - 'candlepin-ca-cert,type=mount,target=/etc/foreman/katello-default-ca.crt'
+      - 'candlepin-client-cert,type=mount,target=/etc/foreman/client_cert.pem'
+      - 'candlepin-client-key,type=mount,target=/etc/foreman/client_key.pem'
+    etc_hosts:
+      "quadlet.example.com": "127.0.0.1"
```

Comment:
Rather environment specific - short term solution?

Reply:
Yes, it's a hack. Capital H.

Reply:
I've left this one as the one "extra" commit in this PR.

Do you want to tackle the "Candlepin should use `localhost` cert if it only listens on localhost" issue here, or in a separate PR?

Reply:
Let's see: https://github.com/ehelms/foreman-quadlet/pull/19